src/test/java/org/openrewrite/java/migrate/joda/JodaTimeRecipeTest.java

```diff
@@ -0,0 +1,206 @@
+        spec
+          .recipe(new JodaTimeRecipe())
+          .parser(JavaParser.fromJavaVersion().classpath("joda-time"));
+    }
+
+    @DocumentExample
+    @Test
+    void migrateSafeVariable() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import org.joda.time.DateTime;
+                            
+              class A {
+                  public void foo() {
+                      DateTime dt = new DateTime();
+                      System.out.println(dt.toDateTime());
+                  }
+              }
+              """,
+            """
+              import java.time.ZonedDateTime;
+
+              class A {
+                  public void foo() {
+                      ZonedDateTime dt = ZonedDateTime.now();
+                      System.out.println(dt);
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void dontChangeClassVariable() {
+        // not supported yet
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import org.joda.time.DateTime;
+                            
+              class A {
+                  public void foo() {
+                      DateTime dt = new DateTime();
+                      System.out.println(dt.toDateTime());
+                      System.out.println(new B().dateTime.toDateTime());
+                  }
+                  public static class B {
+                      DateTime dateTime = new DateTime();
+                  }
+              }
+              """,
+            """
+              import org.joda.time.DateTime;
+
+              import java.time.ZonedDateTime;
+                            
+              class A {
+                  public void foo() {
+                      ZonedDateTime dt = ZonedDateTime.now();
+                      System.out.println(dt);
+                      System.out.println(new B().dateTime.toDateTime());
+                  }
+                  public static class B {
+                      DateTime dateTime = new DateTime();
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void dontChangeIncompatibleType() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import org.joda.time.DateTime;
+
+              class A {
+                  public void foo() {
+                      new B().print(new DateTime()); // print is public method accepting DateTime, not handled yet
+                      System.out.println(new B().dateTime);
+                  }
+              }
+
+              class B {
+                  DateTime dateTime = new DateTime();
+                  public void print(DateTime dateTime) {
+                      System.out.println(dateTime);
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void noUnsafeVar() {
```

Comment:
@sambsnyd  Can you please help, this test was working fine, then I synced with main and started to fail.

JodaTimeRecipeTest > noUnsafeVar() FAILED
    java.lang.IllegalStateException: LST contains missing or invalid type information
    Identifier->Assignment->Block->If->Block->MethodDeclaration->Block->ClassDeclaration->CompilationUnit
    /*~~(Identifier type is missing or malformed)~~>*/dtz
    Identifier->Assignment->Block->Else->If->Block->MethodDeclaration->Block->ClassDeclaration->CompilationUnit
    /*~~(Identifier type is missing or malformed)~~>*/dtz
    https://docs.openrewrite.org/reference/faq#im-seeing-lst-contains-missing-or-invalid-type-information-in-my-recipe-unit-tests-how-to-resolve
        at org.openrewrite.java.Assertions.assertValidTypes(Assertions.java:92)
        at org.openrewrite.java.Assertions.validateTypes(Assertions.java:60)
        at org.openrewrite.java.Assertions$$Lambda$689/0x00007f735034aa00.accept(Unknown Source)
        at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:510)
        at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:132)
        at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:127)
        at org.openrewrite.java.migrate.joda.JodaTimeRecipeTest.noUnsafeVar(JodaTimeRecipeTest.java:133)

Reply:
Fixed it. I was using JavaTemplate for assignment expression incorrectly.